whirlpool/auth.py

```diff
@@ -96,9 +101,8 @@ async def _do_auth(self, refresh_token):
                     elif refresh_token:
                         return await self._do_auth(refresh_token=None)
                     return None
-
-        finally:
-            await session.close()
+        except (aiohttp.ClientConnectionError, asyncio.TimeoutError) as ex:
+            raise ex
```

Comment:
Since we just re-raise the exceptions, I think we can just remove the try/except, or not?